test/regex.jl

```diff
@@ -34,9 +37,10 @@ show(buf, r"")
 # see #10994, #11447: PCRE2 allows NUL chars in the pattern
 @test ismatch(Regex("^a\0b\$"), "a\0b")
 
-# regex match / search string must be a String
-@test_throws ArgumentError match(r"test", GenericString("this is a test"))
```

Comment:
If these are now supposed to work, better keep them and check that they return the correct result.

Reply:
fixed